dbms/src/Server/Server.cpp

```diff
@@ -1545,6 +1544,21 @@ int Server::main(const std::vector<std::string> & /*args*/)
         }
     });
 
+    auto & tmt_context = global_context->getTMTContext();
+    // Resource Control.
+#ifdef DBMS_PUBLIC_GTEST
+    LOG_WARNING(
+        log,
+        "Compiled with ENABLE_TESTS enabled, will use MockLocalAdmissionController, which means tiflash "
+        "ResourceControl will be disabled.");
+#else
+    auto mpp_task_manager = tmt_context.getMPPTaskManager();
+    LocalAdmissionController::global_instance
+        = std::make_unique<LocalAdmissionController>(tmt_context.getKVCluster(), tmt_context.getEtcdClient());
+    LocalAdmissionController::global_instance->registerDeleteResourceGroupCallback(
+        [&mpp_task_manager]() { mpp_task_manager->deleteEmptyScheduler(); });
```

Comment:
Perhaps passing TMTContext as an argument to the TaskScheduler constructor would eliminate the need for code modification.
` TaskScheduler::instance = std::make_unique<TaskScheduler>(config, tmt_context);`

Reply:
`LAC` should be destructed after `MPPTaskManager`(so it can `unregisterDeleteResourceGroup()` successfully).
So it will make things complicated if bind the lifetime of `LAC` with `TaskScheduler`.